src/network/networkd-dhcp4.c

```diff
@@ -583,6 +586,42 @@ static int dhcp4_set_hostname(Link *link) {
         return sd_dhcp_client_set_hostname(link->dhcp_client, hn);
 }
 
+static bool check_promote_secondaries(const char *ifname ) {
+        _cleanup_free_ char *promote_secondaries_sysctl = NULL,*promote_secondaries_path = NULL;
+
+        promote_secondaries_path = strjoin("net/ipv4/conf/", ifname, "/promote_secondaries");
+        if (sysctl_read(promote_secondaries_path, &promote_secondaries_sysctl) >= 0) {
+                unsigned val;
+
+                truncate_nl(promote_secondaries_sysctl);
+                if (safe_atou(promote_secondaries_sysctl, &val) >= 0) {
```

Comment:
if this is a boolean, i.e. either 0 or 1, then please just use parse_boolean() for this. I mean, it parses slightly more than what the kernel will output here, but it's a lot more readable